### PR TITLE
megatools 1.11.0

### PR DESCRIPTION
**Pull request by sidneys**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by chenrui333 — DISMISSED** [review]

File: Formula/megatools.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-  sha256 "8dc1ca348633fd49de7eb832b323e8dc295f1c55aefb484d30e6475218558bdb"
-  license "GPL-2.0"
+  url "https://megatools.megous.com/builds/megatools-1.11.0.20220519.tar.gz"
+  sha256 "b30b1d87d916570f7aa6d36777dd378e83215d75ea5a2c14106028b6bddc261b"
+  license "GPL-2.0-or-later"
 
   livecheck do
     url "https://megatools.megous.com/builds/"
-    regex(/href=.*?megatools[._-]v?(\d+(?:\.\d+)+)\.t/i)
+    regex(/href=.*?megatools[._-]v?(\d{1,3}(?:\.\d{1,3})+)(?:\.\d{8})?\.t/i)
```
Comment (chenrui333) [inline Formula/megatools.rb:10]: cc @samford

Comment (samford) [reply Formula/megatools.rb:10]: If the formula is going to use the full version including the date (`1.11.0.20220519`), then the `livecheck` block shouldn't be modified. The existing regex properly matches versions from filenames like `megatools-1.10.3.tar.gz` as well as `megatools-1.11.0.20220519.tar.gz` and `1.11.0.20220519` is correctly given as the newest version from livecheck.

**Comment by Bo98** [comment]
Hmm, I wonder if this should have `version "1.11.0"` or whether we want the date component. Upstream announce this release as "Megatools 1.11.0". Looks like other package managers are doing the same: https://repology.org/project/megatools/versions

**Comment by SMillerDev** [comment]
Seeing how upstream uses that as identifying information I think it should be part of the version.

**Comment by samford** [comment]
> Seeing how upstream uses that as identifying information I think it should be part of the version.

The [Git repository](https://megous.com/git/megatools) doesn't include the trailing date in the tags (e.g., `1.11.0`, `1.10.2`) and it seems to just be the release date (i.e., `20220519` is the date the tag was created), so I tend to agree with omitting the date from the formula version. If we go this route, I can suggest a new `livecheck` block (we should be checking the `homepage` here anyway).

**Comment by sidneys** [comment]
Hi,

are there any updates on the decision regarding whether to include the date in the version going forward?
I'm happy to adapt the PR in either case.

Kind regards
S

**Comment by chenrui333** [comment]
what's the decision on the `version` field?

**Comment by samford** [comment]
In the interest of moving this forward, can we come to an agreement on the version format?

If we use the full version detected from the filename (i.e., `1.11.0.20220519`), then the `livecheck` block suggestion above should be merged. If we manually specify the version without the date (i.e., `version "1.11.0"`), then I will update my `livecheck` block suggestion to omit an optional trailing date from the filename version.

As above, I'm in favor of using the `1.11.0` format, as this is how the version is referred to on the homepage (e.g., "Megatools 1.11.0 released with new features and bugfixes") and in the [repository](https://megous.com/git/megatools)'s version tags. At present, the trailing date is simply the release date and upstream doesn't appear to be treating it as meaningful information (i.e., it's not part of the version tag, etc.). If upstream starts treating the trailing date as part of the version, then I would agree with using it in the formula version.

**Comment by jonchang** [comment]
[Debian](https://packages.debian.org/sid/source/megatools), [AUR](https://aur.archlinux.org/packages/megatools), and other package managers all seem to be packaging it as 1.11.0, so let's go with that.

**Comment by jonchang** [comment]
From the end of LICENSE:

```text
OpenSSL Exemption
-----------------

In addition, as a special exception, the copyright holders give
permission to link the code of portions of this program with the OpenSSL
library under certain conditions as described in each individual source
file, and distribute linked combinations including the two.

You must obey the GNU General Public License in all respects for all of
the code used other than OpenSSL.  If you modify file(s) with this
exception, you may extend this exception to your version of the file(s),
but you are not obligated to do so.  If you do not wish to do so, delete
this exception statement from your version.  If you delete this
exception statement from all source files in the program, then also
delete it here.
```

**Review by jonchang — APPROVED** [review]

**Comment by samford** [comment]
> [Debian](https://packages.debian.org/sid/source/megatools), [AUR](https://aur.archlinux.org/packages/megatools), and other package managers all seem to be packaging it as 1.11.0, so let's go with that.

@jonchang Unless I'm misinterpreting this, it sounds like you're also in favor of using 1.11.0 as the version but the formula is still using the version with the trailing date (and the `livecheck` block will surface the same). If we want to use `1.11.0`, we'll need to open a follow-up PR to add `version "1.11.0"` and to update the `livecheck` block accordingly. I wasn't able to catch this before it was merged but I already have the `livecheck` block changes sorted, so I can open a PR if you agree.

**Comment by jonchang** [comment]
Gah! Looks like I missed adding an commit to this PR. There's no way to change it now without bumping `version_scheme` which I'm hesitant to do. Maybe it's best to just leave it for now until the next version comes out?

**Comment by samford** [comment]
The next version could be sometime in the near future or it could be as long as four years from now (e.g., between 1.10.2 and 1.11.0) and I don't see any public indicator of which way it could go. For what it's worth, there have only been ~118 installs over the past 30 days and the bottle is ~100 KB, so it may not be a notable burden for affected users (though we would have to perpetually retain `version_scheme` in the formula).

If we don't want to bump `version_scheme`, I could simply create a PR to update the `livecheck` block and add a comment above the `stable` URL to remind us that we want to change the version scheme on the next version (though these comments are sometimes overlooked). The formula version with the appended date would be seen as newer than the livecheck version which should produce a warning annotation on related PRs, so that may also increase our chances of ensuring the version format is updated in the future.

**Pull request closed without merging**